src/Management/src/OpenTelemetryBase/Exporters/Prometheus/PrometheusSerializer.cs

```diff
@@ -26,25 +23,9 @@ internal static partial class PrometheusSerializer
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static int WriteDouble(byte[] buffer, int cursor, double value)
     {
-#if NETCOREAPP3_1_OR_GREATER
-        if (double.IsFinite(value))
-#else
         if (!double.IsInfinity(value) && !double.IsNaN(value))
-#endif
         {
-#if NETCOREAPP3_1_OR_GREATER
-        Span<char> span = stackalloc char[128];
-
-        var result = value.TryFormat(span, out var cchWritten, "G", CultureInfo.InvariantCulture);
-        Debug.Assert(result, $"{nameof(result)} should be true.");
-
-        for (int i = 0; i < cchWritten; i++)
-        {
-            buffer[cursor++] = unchecked((byte)span[i]);
-        }
-#else
             cursor = WriteAsciiStringNoEscape(buffer, cursor, value.ToString(CultureInfo.InvariantCulture));
-#endif
```

Comment:
Looking at this again, I suspect I may have taken the wrong turn here (same in `PullmetricsCollectionManager`). Although technically correct (because the project targeted `netstandard20` initially, making the `true`-blocks effectively dead code), now it seems that the `false` block, which optimizes for `Span<T>` usage, is the preferred path to keep.

Reply:
@hananiel can you take a look at this? it looks like we should be using the code under `#if NETCOREAPP3_1_OR_GREATER`, but that code may not have been getting used before since the project was targeting netstandard 2.0

Reply:
Resolving this conversation to unblock for PR merge. We can revert the change later in a separate PR if needed.